### PR TITLE
feat(app): add splash screen

### PR DESCRIPTION
**Pull request by malashkevich**
### Please read and mark the following check list before creating a pull request:

 - [] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/kittenTricks/blob/master/CONTRIBUTING.md) guide.

 #### Short description of what this resolves:

### CODE REVIEW

File: src/core/appLoader/loadingAnimation.component.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { Animated, Easing, StyleSheet } from 'react-native';
```
Comment (32penkin) [inline src/core/appLoader/loadingAnimation.component.tsx]: New line imports

File: src/core/appLoader/loadingAnimation.component.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    animationValue: new Animated.Value(0),
+    animationCompleted: false,
+  };
+
+  public componentWillReceiveProps(nextProps: Readonly<Props>, nextContext: any): void {
+    if (nextProps.isLoaded && nextProps.isLoaded !== this.props.isLoaded) {
+      this.triggerAnimation();
+    }
+  }
+
```
Comment (32penkin) [inline src/core/appLoader/loadingAnimation.component.tsx:30]: Extra line

File: src/core/appLoader/loadingAnimation.component.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    Animated.timing(this.state.animationValue, {
+      toValue: 1,
+      duration: 700,
+      useNativeDriver: true,
+      easing: Easing.in(Easing.exp),
+    }).start(() => this.setState({ animationCompleted: true }));
+  }
+
+  public render(): React.ReactNode {
+    if (this.state.animationCompleted) {
```
Comment (32penkin) [inline src/core/appLoader/loadingAnimation.component.tsx]: @artyorsh! Bite!

File: src/core/appLoader/loadingAnimation.component.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      return null;
+    }
+
+    const opacity = this.state.animationValue.interpolate({
```
Comment (artyorsh) [inline src/core/appLoader/loadingAnimation.component.tsx]: type

File: src/core/appLoader/loadingAnimation.component.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      inputRange: [0, 1],
+      outputRange: [1, 0],
+    });
+    const transform = [
```
Comment (artyorsh) [inline src/core/appLoader/loadingAnimation.component.tsx]: type

File: src/core/appLoader/loadingAnimation.component.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+
+  private triggerAnimation() {
+    Animated.timing(this.state.animationValue, {
+      toValue: 1,
+      duration: 700,
+      useNativeDriver: true,
+      easing: Easing.in(Easing.exp),
+    }).start(() => this.setState({ animationCompleted: true }));
```
Comment (artyorsh) [inline src/core/appLoader/loadingAnimation.component.tsx]: prefer refactoring to onAnimationCompleted function

File: src/core/appLoader/loadingAnimation.component.tsx (hunk 3 above)
Comment (artyorsh) [inline src/core/appLoader/loadingAnimation.component.tsx]:
```ts
const { animationCompleted } = this.state;

return animationCompleted ? null : this.renderAnimation();
```

File: src/core/appLoader/loadingAnimation.component.tsx (hunk 1 above)
Comment (artyorsh) [inline src/core/appLoader/loadingAnimation.component.tsx]: multiline

File: src/core/appLoader/applicationLoader.component.tsx [hunk 7]
```diff
@@ -28,12 +30,18 @@ type LoadingElement = React.ReactElement<AppLoadingProps>;
  */
 export class ApplicationLoader extends React.Component<Props, State> {
 
+  constructor(props) {
```
Comment (artyorsh) [inline src/core/appLoader/applicationLoader.component.tsx]: `props` arg type

**Review by 32penkin — APPROVED** [review]

File: app.json [hunk 8]
```diff
@@ -2,16 +2,20 @@
   "expo": {
     "name": "kittenTricks",
```
Comment (artyorsh) [inline app.json:3]: rename to Kitten Tricks?

**Review by artyorsh — APPROVED** [review]

**Pull request merged**